lib/std/Build/Step/WriteFile.zig

```diff
@@ -209,6 +266,12 @@ fn make(step: *Step, prog_node: *std.Progress.Node) !void {
             },
         }
     }
+    for (wf.directories.items) |dir| {
+        man.hash.addBytes(dir.source.getPath2(b, step));
+        man.hash.addBytes(dir.sub_path);
+        for (dir.options.exclude_extensions) |ext| man.hash.addBytes(ext);
+        if (dir.options.include_extensions) |incs| for (incs) |inc| man.hash.addBytes(inc);
+    }
```

Comment:
This is a bit iffy and won't necessarily catch up with changes to files within the directory, only changes to the installation options. Directory args to `Step.Run` (which I based this on) has similar issues.

I'm not sure if this is a problem that needs solving or if it's an acceptable tradeoff for not having to recursively hash every single file in a directory tree, but it might be nice to find a solution for this in the future.